Android toolchain support, Fix x86 build errors

copywith9 opened the pull request:
Add apply_android_toolchain, Fix x86 build errors.
Update ogg and flac.

mgeier commented:
Thanks for this PR!

Do the different parts of it depend on each other?
If not, can you please split it into separate PRs?

Are the submodule updates necessary for your other changes?
In general, I'm trying to stay with the official releases, see also #42.

I have never worked with an Android-compatible Rust crate, so please forgive me if I'm asking dumb questions, but did you come up with that setup code from scratch or did you get it from some template?
If latter, it might be helpful to provide a link.

copywith9 commented:
The flutter_rust_bridge module in my Flutter project could not be compiled properly because the platform was not set correctly when cross-compiling cmake.
Then I fixed it and it can now compile normally and is now online. In addition, I updated the dependencies of the flac and ogg submodules.
The modification of bindings.rs is because libflac-sys cannot compile i686 and will report an error indicating a numerical overflow

mgeier commented:
Thanks for the update!

> In addition, I updated the dependencies of the flac and ogg submodules.

But why? Is this necessary for any of the other changes?

> The modification of bindings.rs is because libflac-sys cannot compile i686 and will report an error indicating a numerical overflow

It would be good to fix this, but it sounds like this is independent of the Android changes, right?

Can you please make a separate PR for this?
